### DISCUSSION TITLE
Combat, should we memorize the number of creatures for combats in places protected by guards

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#6883] [General]
In the original game, some places on the adventure map are protected by guards (Graveyard, Dragon city, Shipwreck, Troll Bridge, etc.).
These creatures must be fought and defeated before the location can be accessed.

If during a combat the number of creatures is decreased or increased and the hero runs away without winning the combat, the next time a hero attacks the creatures, their number will be back to the original number as if no creatures had been decimated or added.

Example with the Graveyard.
The hero decimates 50 of the 100 mutant zombies protecting the graveyard and runs away.
The next hero who tries to enter the graveyard will have to fight again 100 mutant zombies.

Example with the Shipwreck.
The ghosts defending the shipwreck will increase in number during the combat before the hero flees.
When a hero attacks the ghosts, the incorporated ghosts will no longer be present and the number of ghosts to fight will be the same as at the beginning of the scenario.

Example with the Dragon City.
The hero manages to eliminate two of the dragons protecting the dragon city before fleeing.
In the next battle, all the dragons will defend the dragon city again.

Should we make it so that the number of creatures in these locations is memorized at the end of a combat where they are not defeated, or should we keep the current operation (as in the original game)?
Should we only consider this in certain cases?
 Especially for ghosts in case their number increases instead of decreases.
This can be proposed as an option.

Personally, I'm fine with either way. ;)

**Comment by zenseii** [comment]
Hi, @LeHerosInconnu . Don't some of these places let you plunder them without knowing if the guards are still there? An example is if you've attacked the graveyard, lost, and then another player attacks it successfully unknowingly to you when you visit it the next time?

**Reply by LeHerosInconnu** [reply]
Hello @zenseii,

If the creatures are defeated by a hero of any player, the place is no longer defended afterwards.
So after that, any hero can enter without fighting.

In the original game, when a hero moves to a protected place, it doesn't mean he fights the creatures, the only way to know is to go and visit the place.
Also, in the original game, unless I'm mistaken, the AI never fights creatures that protect a creature generator first, like dragon city, troll bridge, etc., it is a human player who must defeat them first.

**Edit:**

In the case of the graveyard (and some other places), the player does not know if the guards are still present, and the hero can take a morale penalty if he visits the graveyard and the mutant zombies have already been defeated.

**Reply by zenseii** [reply]
Yes, so we are talking about two different types of adventure map buildings with creature combats.

I'm more curious about graveyards because you should not know beforehand if it has been visited or not.

For other dwellings that's fine. I know Heroes 4 already sort of does this.

**Comment by Alucard648** [comment]
I think, all guardians should respawn, like in H5.

**Reply by LeHerosInconnu** [reply]
Hello @Alucard648,

So just like Heroes 2, or is it different?
I only played the H5 demo briefly.

**Reply by Alucard648** [reply]
Yes, if ypu fail to defeat guardians, on next try you need to redo everything from scratch.

**Comment by Mr-Bajs** [comment]
Isn't the logic to keep fheroes2 as close as the orginal game as possible?

But 1.3 Expansion allows for some changes in this? Decimating guards is a feature that could be intressting for making maps, making the player to do several tries and decimating the guards for a quest or win condition etc. But maybe that feature should be reseved for any new assets that might be added, if it should be implemented at all?

**Reply by LeHerosInconnu** [reply]
Hello @Mr-Bajs,

> Isn't the logic to keep fheroes2 as close as the orginal game as possible?

Yes, up to version 1.0, after that the game can be improved, refined, polished. :)

> Decimating guards is a feature that could be intressting for making maps, making the player to do several tries and decimating the guards for a quest or win condition etc.

Even in the current game, it could bring new ways of doing things.
For example, making coordinated attacks with several heroes to defeat the guardians of a location.

**Comment by oleg-derevenetz** [comment]
I suggest not to pull the game logic back and forth and just reproduce the OG behavior if possible (if logic is clear and consistent and there is no obvious bug in OG). I just fixed another long-standing bug opened by you (#5651) and now you want the opposite.

**Comment by LeHerosInconnu** [comment]
It's okay the way it works currently.
I close the discussion.
